docs/chapters/traversal/index.md

```diff
@@ -0,0 +1,178 @@
+# Understand Jina Recursive Document Representation
+
+In Jina, each `Document` is represented as a recursive representation (tree).
+
+A rooted recursive representation has a root node and every node has X children.
+In Jina, the root node is the document itself, while the *left* & *right* children are referred to as *chunks* and *matches* respectively.
+
+![rooted-binary-tree](img/rooted-binary-tree.png)
+
+The above image illustrates a minimal document structure: A document (root node) consists of two child nodes: *chunks* and *matches*.
+`chunks`s are a sequence of `Document`s which is attached to the root document with a higher `granularity` degree. `Matches` is a sequence of `Document`s  which are semantically related to the root document.
+We'll dive into these concepts in this document:
+
+- [Chunks](#chunks)
+- [Matches](#matches)
+- [Let's go deeper: Recursive Document Representation](#lets-go-deeper-recursive-document-representation)
+- [Document Traversal with traversal paths](#document-traversal-with-traversal-paths)
+
+
+## Chunks
+
+Each Jina `Document` (potentially) consists of a list of `Chunk`s. A `Chunk` is a small semantic unit of a `Document`, like a sentence or a 64x64 pixel image patch.
+
+Think about these use cases: you want to search a document at a specific `granularity` level, e.g. a sentence or a paragraph.
+Or your query consists of multiple modalities, such as your query consist of a piece of text together with an image.
```

Comment:
"Or your query consists of multiple modalities, such as a piece of text together with an image."